raster/src/main/scala/geotrellis/raster/reproject/TileReprojectMethods.scala

```diff
@@ -0,0 +1,31 @@
+package geotrellis.raster.reproject
+
+import geotrellis.raster._
+import geotrellis.proj4._
+import geotrellis.raster.resample.ResampleMethod
+import geotrellis.vector.Extent
+
+trait TileReprojectMethods[T] extends MethodExtensions[T] {
```

Comment:
If `T <: CellGrid` the reproject methods could be returning `Raster[T]`, which is nicer API.